tests/dataloaders/test_datasplitter.py

```diff
@@ -40,6 +40,38 @@ def test_datasplitter_shuffle():
     )
 
 
+def test_datasplitter_external():
```

Comment:
Ideally we should also be testing for error cases (overlapping indexes, missing input arguments, etc). You can do this with `with pytest.raises`.